application-ldapuserimport-api/src/main/java/com/xwiki/ldapuserimport/internal/job/DefaultLDAPGroupImportJob.java

```diff
@@ -0,0 +1,151 @@
+import org.xwiki.contrib.ldap.XWikiLDAPConfig;
+import org.xwiki.contrib.ldap.XWikiLDAPException;
+import org.xwiki.contrib.ldap.XWikiLDAPSearchAttribute;
+import org.xwiki.job.event.status.JobProgressManager;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.model.reference.EntityReferenceSerializer;
+import org.xwiki.wiki.descriptor.WikiDescriptorManager;
+
+import com.xpn.xwiki.XWiki;
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.XWikiException;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xwiki.ldapuserimport.LDAPGroupImportManager;
+import com.xwiki.ldapuserimport.LDAPUserImportManager;
+import com.xwiki.ldapuserimport.internal.XWikiLDAPUtilsHelper;
+import com.xwiki.ldapuserimport.job.AbstractLDAPGroupImportJob;
+
+/**
+ * Default implementation of the {@link AbstractLDAPGroupImportJob}.
+ *
+ * @version $Id$
+ * @since 1.4
+ */
+@Component
+@Named(AbstractLDAPGroupImportJob.JOB_TYPE)
+public class DefaultLDAPGroupImportJob extends AbstractLDAPGroupImportJob
+{
+    @Inject
+    private EntityReferenceSerializer<String> stringEntityReferenceSerializer;
+
+    @Inject
+    private JobProgressManager jobProgressManager;
+
+    @Inject
+    private LDAPDocumentHelper ldapDocumentHelper;
+
+    @Inject
+    private LDAPGroupImportManager ldapGroupImportManager;
+
+    @Inject
+    private LDAPUserImportManager ldapUserImportManager;
+
+    @Inject
+    private Provider<XWikiLDAPConfig> xWikiLDAPConfigProvider;
+
+    @Inject
+    private Provider<XWikiContext> xWikiContextProvider;
+
+    @Inject
+    private WikiDescriptorManager wikiDescriptorManager;
+
+    @Override
+    public String getType()
+    {
+        return JOB_TYPE;
+    }
+
+    @Override
+    protected void runInternal() throws Exception
+    {
+        // Start by getting the list of available LDAP groups from the LDAP directory
+        jobProgressManager.startStep(this, "Retrieve the list of importable groups");
+        Map<String, List<XWikiLDAPSearchAttribute>> importableGroups = ldapGroupImportManager.getImportableGroups(
+            request.getLDAPGroupSearchDN(), request.getLDAPGroupSearchFilter(), request.getLDAPGroupSearchAttributes());
+        jobProgressManager.endStep(this);
+
+        // Exclude every LDAP group already mapped to an XWiki group
+        jobProgressManager.startStep(this, "Exclude existing linked groups");
+        Map<String, Set<String>> mappings = xWikiLDAPConfigProvider.get().getGroupMappings();
+        for (Map.Entry<String, Set<String>> mapping : mappings.entrySet()) {
+            for (String existingLDAPGroup : mapping.getValue()) {
+                importableGroups.remove(existingLDAPGroup);
+            }
+        }
+        jobProgressManager.endStep(this);
+
+        // With the remaining groups, compute their XWiki group name, create a document, and register them as bindings
+        jobProgressManager.startStep(this, "Create new groups and register their mappings");
```

Comment:
Would be nice to report the progress for each imported group with:
```
jobProgressManager.pushLevelProgress(importableGroups.size(), this);
...
    jobPogressManager.startStep(this);
    ...
    jobPogressManager.endStep(this);
...
jobProgressManager.popLevelProgress(this);
```

Reply:
Indeed ; thanks for the idea, added in 6f635b1